docs/2.7.0/docs/app-dev/explicit-contract-disclosure.rst

```diff
@@ -0,0 +1,270 @@
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _explicit-contract-disclosure:
+
+Explicit Contract Disclosure (Experimental)
+###########################################
+
+Canton 2.7 introduces explicit contract disclosure as a feature
+that allows you to seamlessly delegate contract read rights to a non-stakeholder using off-ledger data distribution.
+
+Toggle the ``explicit-disclosure-unsafe`` flag in the participant configuration as shown below
+to use disclosed contracts in command submission by means of explicit contract disclosure.
+
+.. note::  This feature is **experimental** and **must** not be used in production environments.
+
+::
+
+    participants {
+        participant {
+            ledger-api.explicit-disclosure-unsafe = true
+        }
+    }
+
+Contract Read Delegation
+------------------------
+
+Contract read delegation allows a party to acquire read rights during
+command submission over a contract of which it is neither a stakeholder nor an informee.
+
+As an example application where read delegation could be used,
+let's consider a simplified trade between two parties.
+In this example, party **Seller** owns a unit of Digital Asset ``Stock`` issued by the **StockExchange** party.
+As the issuer of the stock, **StockExchange** also publishes the stock's ``PriceQuotation`` as public data,
+which can be used for settling trades at correct market value. The **Seller** announces its offer
+to sell his stock publicly by creating an ``Offer`` contract that can be exercised by anyone that
+can pay the correct market value in terms of ``IOU`` units.
+
+On the other side, party **Buyer** owns an ``IOU`` with 10 monetary units, which he wants to
+use to acquire **Seller** 's stock.
+
+Below we outline the Daml templates used to model the above-mentioned trade.
+
+::
+
+    template IOU
+      with
+        issuer: Party
+        owner: Party
+        value: Int
+      where
+        signatory issuer
+        observer owner
+
+        choice IOU_Transfer: ()
+          with
+            target: Party
+            amount: Int
+          controller owner
+          do
+            -- Check that the transferred amount is not higher than the current IOU value
+            assert (value >= amount)
+            create this with issuer = issuer, owner = target, value = amount
+            -- No need to create a new IOU for owner if the full value is transferred
+            if value == amount then pure ()
+            else void $ create this with issuer = issuer, owner = owner, value = value - amount
+            pure ()
+
+    template Stock
+      with
+        issuer: Party
+        owner: Party
+        stockName: Text
+      where
+        signatory issuer
+        observer owner
+
+        choice Stock_Transfer: ()
+          with
+            newOwner: Party
+          controller owner
+          do
+            create this with owner = newOwner
+            pure ()
+
+    template PriceQuotation
```

Comment:
This example requires a one line comment as to how the `PriceQuotation` is intended to be used. I.e. while active it witnesses the "current fair price" and can be used as input to [market orders](https://www.investopedia.com/terms/m/marketorder.asp)